crud/common/call.lua

```diff
@@ -52,39 +32,42 @@ local function call_impl(vshard_call, func_name, func_args, opts)
         end
     end
 
-    local nodes_count = utils.table_count(replicasets)
-    local channel = fiber.channel(nodes_count)
-
+    local futures = {}
+    local call_opts = {is_async = true}
     for _, replicaset in pairs(replicasets) do
-        fiber.create(
-            call_on_replicaset, replicaset, channel, vshard_call, func_name, func_args, {
-                timeout = timeout,
-            }
-        )
+        local future = replicaset[vshard_call](replicaset, func_name, func_args, call_opts)
+        futures[replicaset.uuid] = future
     end
 
     local results = {}
-
-    for _ = 1, channel:size() do
-        local res = channel:get()
-
-        if res == nil then
-            if channel:is_closed() then
-                return nil, CallError:new("Channel is closed")
-            end
-
-            return nil, CallError:new("Timeout was reached")
+    local deadline = fiber_clock() + timeout
+    for replicaset_uuid, future in pairs(futures) do
+        local wait_timeout = deadline - fiber_clock()
+        if wait_timeout < 0 then
+            wait_timeout = 0
```

Comment:
Should we perform `future:wait_result(0)` call anyway, or simply return error?

Reply:
You should return special kind of box.error in this case box.error.timeout. I don't want to overcomplicate this logic - let's simply pass 0 to timeout.